tests/lib/rules/class-methods-use-this.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @fileoverview Tests for class-methods-use-this rule.
+ * @author Patrick Williams
+ * @copyright 2016 Patrick Williams. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/class-methods-use-this");
+const RuleTester = require("../../../lib/testers/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+
+ruleTester.run("class-methods-use-this", rule, {
+    valid: [
+        {code: "class A { constructor() {} }", parserOptions: { ecmaVersion: 6 }},
+        {code: "class A { foo() {this} }", parserOptions: { ecmaVersion: 6 }},
+        {code: "class A { foo() {this.bar = 'bar';} }", parserOptions: { ecmaVersion: 6 }},
+        {code: "class A { foo() {bar(this);} }", parserOptions: { ecmaVersion: 6 }},
+        {code: "class A { foo() { if(true) { return this; } } }", parserOptions: { ecmaVersion: 6 }},
+        {code: "class A { static foo() {} }", parserOptions: { ecmaVersion: 6 }},
+        {code: "class A { constructor() {} }", parserOptions: { ecmaVersion: 6 }},
+        {code: "({ a(){} });", parserOptions: { ecmaVersion: 6 }},
+        {code: "class A { foo() { () => this; } }", parserOptions: { ecmaVersion: 6 }}
```

Comment:
What about this? `{code: "({ a: function () {} });"}`? My assumption is that should be valid as well since it's not a class.